src/FibonacciHeap.java

```diff
@@ -101,11 +101,52 @@ private void consolidate() {
 	 * 
 	 */
 	public void decreaseKey(HeapNode x, int diff) {
-		// TODO: By Itay.
-		// Please look at the comment in detatchNode
-		return; // should be replaced by student code
+		x.key = x.key - diff;
+		if (x.parent == null|| x.parent.key<= x.key) { 
+			// The decrease is legal with the heap logic 
+			return;
+		}
+		else  {
+			cascadeCut(x);
+		}
+		return; 
+	}
+	/**
+	 * 
+	 * Handle the cutting process.
+	 *
+	 */
+	private void cascadeCut(HeapNode x) {
+		if (x.parent == null) 
+			//Terminate cascade when we reach root
+			return;
+		HeapNode parent = x.parent;
+		if (x.parent.child == x) {
+			if (x.next == x) {
+				// Only child
+				x.parent.child = null;
+			} else {
+				// Connect parent to sibling
+				x.parent.child = x.next;
+			}
+		}
```

Comment:
Defined a variable and then didn't use it (parent)
```suggestion
		if (parent.child == x) {
			if (x.next == x) {
				// Only child
				parent.child = null;
			} else {
				// Connect parent to sibling
				parent.child = x.next;
			}
		}
```